Description: Upstream Changeset 2609+2612
  stop logging postgres database password in gpgsql connection errors.
Origin: http://wiki.powerdns.com/trac/changeset/2609/ + http://wiki.powerdns.com/trac/changeset/2612/

--- a/modules/gpgsqlbackend/spgsql.cc
+++ b/modules/gpgsqlbackend/spgsql.cc
@@ -26,8 +26,12 @@
   if(!port.empty())
     d_connectstr+=" port="+port;
 
-  if(!password.empty())
+  d_connectlogstr=d_connectstr;
+
+  if(!password.empty()) {
+    d_connectlogstr+=" password=<HIDDEN>";
     d_connectstr+=" password="+password;
+  }
   
   ensureConnect();
 }
@@ -55,7 +59,7 @@
 
   if (!d_db || PQstatus(d_db)==CONNECTION_BAD) {
     try {
-      throw sPerrorException("Unable to connect to database, connect string: "+d_connectstr);
+      throw sPerrorException("Unable to connect to database, connect string: "+d_connectlogstr);
     }
     catch(...) {
       if(d_db)
--- a/modules/gpgsqlbackend/spgsql.hh
+++ b/modules/gpgsqlbackend/spgsql.hh
@@ -26,6 +26,7 @@
   void ensureConnect();
   PGconn* d_db; 
   string d_connectstr;
+  string d_connectlogstr;
   PGresult* d_result;
   int d_count;
   static bool s_dolog;
